[llvm][Bitcode][ObjC] Fix order of setter/getter argument to DIObjCProperty constructor

Michael137 opened the pull request:
Depends on:
* https://github.com/llvm/llvm-project/pull/165401

We weren't testing `DIObjCProperty` roundtripping. So this was never caught.

The consequence of this is that the `setter:` would have the getter name
and `getter:` would have the setter name.

Michael137 committed:
[llvm][DebugInfo][ObjC] Fix argument oreder of setter/getter to DIObjDIObjCProperty constructor

Depends on https://github.com/llvm/llvm-project/pull/165373

This caused the `DW_AT_APPLE_property_(setter|getter)` to be inverted when compiling from LLVM IR.

OCHyams commented on llvm/lib/Bitcode/Reader/MetadataLoader.cpp:
```diff
@@ -2323,7 +2323,7 @@ Error MetadataLoader::MetadataLoaderImpl::parseOneMetadata(
                         (Context, getMDString(Record[1]),
                          getMDOrNull(Record[2]), Record[3],
-                         getMDString(Record[4]), getMDString(Record[5]),
+                         getMDString(Record[5]), getMDString(Record[4]),
```
Possibly worth adding a comment, or using variable names or something here? On the other hand maybe that doesn't really add much value - I'll leave it up to you.

OCHyams reviewed (approved):
Not familiar with Objc but the change is simple and makes sense, LGTM

Michael137 committed:
[llvm][Bitcode][ObjC] Fix order of setter/getter argument to DIObjCProperty constructor

Depends on:
* https://github.com/llvm/llvm-project/pull/165401

We weren't testing `DIObjCProperty` roundtripping. So this was never caught.

The consequence of this is that the `setter:` would have the getter name
and `getter:` would have the setter name.

Michael137 committed:
fixup! add comments
